q2_fmt/_peds.py

```diff
@@ -398,6 +400,21 @@ def _filter_associated_reference(reference_series, metadata, time_column,
                            ' timepoint value have an associated reference.'
                            ' IDs where missing references were found:'
                            ' %s' % (tuple(nan_references),))
+    available_references = (used_references.isin(ids_with_data))
+    if not available_references.all():
+        if filter_missing_references:
+            used_references = used_references[available_references]
+        else:
+            raise KeyError('References included in the metadata are missing'
+                           ' from the feature table. Please make sure all'
+                           ' references included in the metadata are also'
+                           ' present in the table.'
+                           ' Missing references: %s'
+                           % list(used_references[~available_references]
+                                  .unique()))
+
+    used_references = used_references[available_references]
+    metadata = metadata.filter(items=used_references.index, axis=0)
```

Comment:
Similarly to the above, would call this `metadata_df` or at least type the parameter as `pd.DataFrame` in the signature.